src/hooks/useCredsManager.ts

```diff
@@ -3,150 +3,167 @@ import { useCallback, useEffect } from "react";
-		getCurrentUserCredentials,
+		fetchCredentials,
 		clearCredentials,
 		region,
+		isUserAwsAccountConnected,
+		identityPoolId,
+		fetchTokens,
+		refreshTokens,
+		userPoolId,
+		userPoolClientId,
 		authTokens,
-		setAuthTokens,
-		onLogout,
-		handleCurrentSession,
-		isUserAwsAccountConnected
-	} = useAmplifyAuth();
-	const { locationClient, createLocationClient, iotClient, createIotClient, resetStore: resetAwsStore } = useAws();
-	const { attachPolicy } = useAwsIot();
-	const { mapProvider: currentMapProvider } = useAmplifyMap();
+		setAuthTokens
+	} = useAuth();
+	const {
+		locationClient,
+		createLocationClient,
+		iotClient,
+		createIotClient,
+		resetStore: resetClientStore
+	} = useClient();
+	const { attachPolicy } = useIot();
+	const { mapProvider: currentMapProvider } = useMap();
 	const shouldClearCredentials = localStorage.getItem(SHOULD_CLEAR_CREDENTIALS) === "true";
 
-	const clearCredsAndLocationClient = useCallback(() => {
+	const clearCredsAndClients = useCallback(() => {
 		clearCredentials();
-		resetAwsStore();
-	}, [clearCredentials, resetAwsStore]);
+		resetClientStore();
+	}, [clearCredentials, resetClientStore]);
 
 	if (shouldClearCredentials || (!!credentials && !credentials?.identityId)) {
 		localStorage.removeItem(SHOULD_CLEAR_CREDENTIALS);
-		clearCredsAndLocationClient();
+		clearCredsAndClients();
 	}
 
+	const refreshCredentials = useCallback(async () => {
+		if (credentials && credentials.authenticated) {
+			(async () => {
+				await refreshTokens();
+			})();
+		}
+
+		clearCredsAndClients();
+	}, [clearCredsAndClients, credentials, refreshTokens]);
+
 	/* Fetch the current user credentials */
 	useEffect(() => {
-		if (credentials?.identityId && credentials?.expiration) {
+		if (credentials && credentials.expiration) {
 			const now = new Date();
 			const expiration = new Date(credentials.expiration);
 
 			if (now > expiration) {
-				/* If the credentials are expired, clear them and the location client */
-				clearCredsAndLocationClient();
+				/* If the credentials are expired, refresh them */
+				refreshCredentials();
 			} else {
-				/* If the credentials are not expired, set the refresh interval/timeout */
-				interval && clearInterval(interval);
+				/* If the credentials are not expired, set the refresh timeout */
 				timeout && clearTimeout(timeout);
-
-				if (credentials.authenticated) {
-					/* If the credentials are authenticated, set the refresh interval */
-					interval = setInterval(() => {
-						handleCurrentSession(resetAwsStore);
-					}, 10 * 60 * 1000);
-				} else {
-					/* If the credentials are not authenticated, set the refresh timeout */
-					timeout = setTimeout(() => {
-						clearCredsAndLocationClient();
-					}, differenceInMilliseconds(new Date(credentials.expiration || 0), new Date()));
-				}
+				timeout = setTimeout(() => {
+					refreshCredentials();
+				}, differenceInMilliseconds(new Date(credentials.expiration || 0), new Date()) - 300);
```

Comment:
The previous logic didn't have the "- 300", why is that needed now?

Reply:
Credentials are refreshed 5 mins before expiration